OneCommunityGlobal/HighestGoodNetworkApp
# Kaixiang - Create 24/48/72 hours button on team member tasks tab

### **Description:**
 Implement the 24/48/72h buttons to show the time logs during this period. 
Ex: If the user click 24hours button on May 6, it will fetch time log from May 5 - May 6 (May 6 included).
     If the user click 48hours button on May 6, it will fetch time log from May 4 - May 6 (May 6 included).
     and so on...

### **Mainly change explained:**
1. Restructure the code in `teamMemberTasks `component. 
2. Created 24/48/72h buttons, view/hide icon, delete icon.
3. Optimize the logic of fetching timelogs data. 
4. Loading animation at the first render of those three buttons. (Fetching data from database).
5. No change on backend.

### **UI explained:**
![new UI](https://user-images.githubusercontent.com/98206107/236638795-600e10cb-1b29-4a38-be90-d1af01ed0764.JPG)
1. 24/48/72 buttons: Original status is colored text and border. When selected it turns into colored background and white text.
2. Eye/Eye-Slash icon: It will not show up until you select one of the buttons. Click to show/hide timelogs.
3. Delete icon: Clear the view of time logs, and view of eye/delete icon.

### **How to test**
1. Try three buttons see if timelogs can be correctly displayed.
2. Try the eye/delete icon.
3. Also try other functionalities in this component to make sure everything working well after the restructuring of the code.
4. If you want to display the timelog that you just submitted, you just need to refresh the page.
